twoliter/embedded/Makefile.toml

```diff
@@ -354,11 +354,29 @@ dependencies = ["setup"]
 script_runner = "bash"
 script = [
 '''
+declare -a CARGO_MANIFESTS
+declare -a CARGO_DIGESTS
+mkdir -p .cargo/vendor
+if [ ! -f ".cargo/config.toml" ]; then
+    touch .cargo/config.toml
+fi
 for ws in sources variants .; do
   [ -s "${BUILDSYS_ROOT_DIR}/${ws}/Cargo.toml" ] || continue
-  cargo fetch --locked --manifest-path "${BUILDSYS_ROOT_DIR}/${ws}/Cargo.toml"
+  CARGO_MANIFESTS+=(--sync "${BUILDSYS_ROOT_DIR}/${ws}/Cargo.toml")
+  CARGO_DIGESTS+=("$(sha256sum "${BUILDSYS_ROOT_DIR}/${ws}/Cargo.lock" | awk '{ print $1 }')")
 done
-
+DIGEST="$(echo ${CARGO_DIGESTS[@]} | sha256sum | awk '{ print $1 }')"
+MARKER_FILE=".cargo/vendor-${DIGEST}.marker"
+if [ -f "${MARKER_FILE}" ]; then
+    echo "no changes detected in cargo projects, skipping cargo vendor"
+else
+    cargo vendor --versioned-dirs --respect-source-config --locked ${CARGO_MANIFESTS[@]} .cargo/vendor > .cargo/config.toml 2> .cargo/vendor.log || {
```

Comment:
Can you check that we don't clobber settings in `.cargo/config.toml`?

You mentioned in person that `--respect-source-config` probably does this and I think you're right, but we should test it.